### ISSUE TITLE
Usage

### ISSUE THREAD

**Issue by louis49** [halunka/meteor-videobackground#1]
Hello

I don't understand the usage. Where can i set the video url ? 

Tks.

**Comment by halbgut** [comment]
I'm sorry, we gotta add extend that read me. It depends, are you using iron:router?

**Comment by stefan-wiesner** [comment]
Hey. Can you please tell us how to use it and where to put which code?

For me the vbgOptions have no impact. All values stay empty if the html is rendered.

<div class="videobackground">
    <div class="videobackground__poster"></div>

  <iframe style="margin-top: -436px; margin-left: -801px;" id="NaN" class="videobackground__frame videobackground__frame--block" src="//player.vimeo.com/video/?loop=1&amp;title=0&amp;badge=0&amp;byline=0&amp;portrait=0&amp;api=1&amp;player_id=NaN" frameborder="0" height="925px" width="1651.125px"></iframe></div>

Ok. And how to add a poster image? 

Nowhere in your  code an id is set. Where do I set the vimeo ID?

Thanks for your help

**Comment by halbgut** [comment]
Hi @stefan-wiesner,
`vbgOptions` is an object you should write to your data-context. So if you're using `iron:router` you'd do something like this:

``` js
Router.route('/someRoute', {
  name: 'someRoute',
  data: {
    id: '140850530', // for https://vimeo.com/channels/staffpicks/140850530
    type: 'vimeo',
    aspectRatio: 1.785,
  }
})
```

Poster images aren't really implemented in this package. I'm working sitting on some uncommited changes that implement that, though. I'll probably release it tonight.
What you can do right now is add a background-image to `.videobackground`. That'll pretty much do the same thing. You'll just get a weird flicker, which the update will get rid of.
The `id` attribute is actually missing. I totally missed that, sorry.

**Comment by halbgut** [comment]
Hi again, @stefan-wiesner,

I've published the update. There's still a flicker with the poster. But I don't think I can avoid it, since it's an iframe.

**Comment by stefan-wiesner** [comment]
Hey. 

Thanks for the fast response. Did an update and put the code into the router. 

Now I get a fullscreen Vimeo message that the embed code is wrong because no parameter seems to be taken into account. 

My router:

```
 Router.route('home', {
        path: '/'
         data: {
           id: '139338289',
           player_id: '139338289',
           loop: 0,
           type: 'vimeo',
           volume: 0,
           poster: '/img/surfer.jpg',
           aspectRatio: 1.785,
           autoSize: 1
         }
});
```

home.html
....

```
{{ > videobackground vbgOptions }} 
```

....

Also tried this in home.js:

```
Template.videobackground.rendered = function(){
 Blaze.renderWithData('template', {
   vbgOptions: {
     id: '139338289', // for https://vimeo.com/channels/staffpicks/140850530
           player_id: '139338289',
           loop: 0,
           type: 'vimeo',
           volume: 0,
           poster: '/img/surfer.jpg', // An image inside your public folder
           aspectRatio: 1.785,
           autoSize: 1
   }
 }, document.body)
```

and also 

```
Template.videobackground.onRendered(function () {
  renderVideo(this.firstNode, _.extend(
    {
      id: '139338289',
      loop: 1,
      type: 'vimeo',
      volume: 0,
      aspectRatio: 1.785,
      autoSize: 1,
      setTimeout: 2000
    },
    this.data)
  )
});
```

No matter what the values are not filled in in the iframe code. I'm on meteor 1.2.0.2 maybe thats the prob. 

Sorry to bug you but I would really like to see this in action.

cheers

Stefan

**Comment by halbgut** [comment]
Sorry, I was kinda tired when I wrote my last comment. I made an error in it. It should actually be

``` js
Router.route('/someRoute', {
  name: 'someRoute',
  data: {
    vbgOptions: {
      id: '140850530', // for https://vimeo.com/channels/staffpicks/140850530
      type: 'vimeo',
      aspectRatio: 1.785,
    }
  }
})
```

I forgot to put it inside the `vbgOptions` object. With that, your first example should work.

There are multiple problems with your second example. First, a conceptual one. At the point where your callback will be called, my package will already have tried and failed to render the video. Once the template is rendered, you cannot change the data-context anymore. Then your usage and understanding of `Blaze.renderWithdata`; It renders a whole template. Iron does that for you when you pass a name parameter. The first parameter to `Blaze.renderWithdata` must be an instance of `Blaze.Template` or `Blate.View. So if you wanted to render the`videobackground`template (again), you'd have to pass`Template.videobackground`as the first parameter. The`player_id`option is ignored. Also, all options other than`id` are optional. Their defaults are the options I pass inside the example in the readme file.

With the third example you're probably getting an `undefined is not a function error`. That's because the function is local to my package.

I've tested it with Meteor 1.2 and with your options. That worked.

Please tell me if anything is unclear in my comment or if you have another question! :)

**Comment by stefan-wiesner** [comment]
Hey. 

I don't get it sorry. Maybe I am too stupid or too tired :-) Haven't used renderWithData before and I am no meteor crack but it seems staight forward and should have had worked... 

So. As far as I understood the router ships the options to the template which renders it with the data/options. Fine. 

But for some reason my options are totally ignored. And that's whats been rendered into the page:

```
<div class="videobackground">
    <div style="background-image: url(&quot;undefined&quot;);" class="videobackground__poster videobackground__poster--hidden"></div>
  <iframe style="margin-top: -420px; margin-left: -772px;" id="NaN" class="videobackground__frame" src="//player.vimeo.com/video/?loop=1&amp;title=0&amp;badge=0&amp;byline=0&amp;portrait=0&amp;api=1&amp;player_id=NaN" frameborder="0" height="893px" width="1594.0049999999999px"></iframe></div>
```

It's calculating the width/height and position but the id for example is not there.

I have a fairly simple setup:

router:

```
Router.map(function() {

    this.route('home', {
        path: '/',
        data: {
          vbgOptions: {
            loop: 0,
            id: '139338289',
            poster: '/img/poster.jpg'
          }
        }
    });
});
```

home.html

```
<template name="home">
   {{ > videobackground vbgOptions }}
</template>
```

Nothing in home.js because the template should be rendered with the data from the router.

If I would like to re-render it I would do something like that:

```
 Blaze.renderWithData('Template.videobackground',                                                                     
    {      
       id: '140850530'      
    },                                                                                                        
    document.body)      
```

I've tried trillions of combinations without luck. Can you tell me what I am doing wrong?

**Comment by halbgut** [comment]
Hi!

I've put up a demo using your code [on this meteorpad](http://meteorpad.com/pad/7GLpSycx33NWDbCcK/videobackground-demo). `Router.map` is deprecated, as far as I know (I'm not an iron:router user).

You might have an old version of the package installed. Maybe try running `meteor update`.

Hope this helps! :)

**Comment by stefan-wiesner** [comment]
Hey. Thanks for you continuous help. 

I managed to get it working with your example. 

I reinstalled basically everything and did a meteor update but it did not work. Then I did some modifications to your file and it worked. Then I undind everything and it is still working. 

I have no clue why it was not working with nothing changed on my side. Maybe because its inside a docker container it does not register the filechange and stayed with a pregenerated oder version. 

Don't know but now it works. 

Thanks alot for you endurance to help me ;-)
